### PR TITLE
cover filter navigator with tests

### PR DESCRIPTION
**Pull request by hiagolcm**
#### What problem is this solving?

- To cover the Filter navigator code with tests.
- It is not the intention to cover edge cases. We will do it later. We are focusing on the code that already exists.

#### How to test it?

`yarn test`

[Workspace](https://hiago--storecomponents.myvtex.com/)

#### Screenshots or example usage:

image

image

### CODE REVIEW

File: react/__tests__/FilterNavigator.test.js [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
     setQuery: mockSetQuery,
     getSettings: () => ({}),
+    query: {},
   }))
+
+  useDevice.mockImplementation(() => ({ isMobile: false }))
 })
 
 describe('<FilterNavigator />', () => {
   const renderComponent = (customProps = { query: 'clothing' }) => {
```
Comment (thalytafabrine) [inline react/__tests__/FilterNavigator.test.js:33]:
```suggestion
  const renderComponent = (customProps = { query: 'clothing', map: 'c' }) => {
```
so you don't have to check the map below :)

File: react/__tests__/FilterNavigator.test.js [hunk 2]
```diff
@@ -45,15 +51,405 @@ describe('<FilterNavigator />', () => {
...
+  it('should allow to add a root category when there is no categories selected', () => {
+    const { getByText } = renderComponent({ query: 'fulltextterm', map: 'ft' })
+
+    fireEvent.click(getByText(/Eletrônicos/))
+
+    expect(mockNavigate).toHaveBeenCalledTimes(1)
+    expect(mockNavigate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        to: '/fulltextterm/eletronicos',
+        query: 'map=ft',
```
Comment (thalytafabrine) [inline react/__tests__/FilterNavigator.test.js:124]: in this case, shouldn't the map be `ft,c` or something like this?

File: react/__tests__/FilterNavigator.test.js [hunk 3]
```diff
@@ -45,15 +51,405 @@ describe('<FilterNavigator />', () => {
...
+
+  it('should navigate to root category on key down', () => {
+    const { getByTestId } = renderComponent({
+      query: 'fulltextterm',
+      map: 'ft',
+    })
+
+    fireEvent.keyDown(getByTestId('root-category-eletronicos'), {
+      key: 'Enter',
+      code: 'Enetr',
```
Comment (thalytafabrine) [inline react/__tests__/FilterNavigator.test.js:233]:
```suggestion
      code: 'Enter',
```
there are some more like this below

File: react/__tests__/components/FacetItem.test.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    expect(mockNavigate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        ...clonedFacet,
+        title: mockProps.facetTitle,
+      }),
+      mockProps.preventRouteChange
+    )
+  })
+
+  it('should give a feedback on select/unselct', () => {
```
Comment (thalytafabrine) [inline react/__tests__/components/FacetItem.test.js:76]:
```suggestion
  it('should give a feedback on select/unselect', () => {
```

**Review by thalytafabrine — COMMENTED** [review]
also, the file for the PriceRange has a typo:  `react/__tests__/components/PrceRange.test.js`

**Review by camilavcoutinho — APPROVED** [review]

**Review by thalytafabrine — APPROVED** [review]
